fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java

```diff
@@ -0,0 +1,92 @@
+package com.github.dedis.popstellar.testutils;
+
+import com.github.dedis.popstellar.model.objects.security.KeyPair;
+import com.github.dedis.popstellar.model.objects.security.MessageID;
+import com.github.dedis.popstellar.model.objects.security.PoPToken;
+import com.github.dedis.popstellar.model.objects.security.PublicKey;
+import com.github.dedis.popstellar.model.objects.security.Signature;
+import com.github.dedis.popstellar.model.objects.security.privatekey.PlainPrivateKey;
+
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters;
+import org.bouncycastle.crypto.params.Ed25519PublicKeyParameters;
+
+import java.util.Base64;
+import java.util.Random;
+
+/**
+ * This class provides pseudo random generators for base64 url data
+ *
+ * <p>This uses a fixed seed. So two test runs will have the same generated data This is important
+ * when trying to extract the cause ofa failing test
+ */
+public class Base64DataUtils {
+
+  private static final long SEED = 2022;
+  private static final Random RANDOM = new Random(SEED);
+
+  private static final int KEY_LENGTH = 32;
+  private static final int MSG_ID_LENGTH = 32;
+  private static final int SIGNATURE_LENGTH = 54;
+
+  /** @return a pseudo randomly generated message id */
+  public static MessageID generateMessageID() {
+    byte[] msgId = new byte[MSG_ID_LENGTH];
+    RANDOM.nextBytes(msgId);
+    return new MessageID(Base64.getUrlEncoder().encodeToString(msgId));
+  }
+
+  /** @return a pseudo randomly generated message id different from the one provided */
+  public static MessageID generateMessageIDOtherThan(MessageID other) {
+    while (true) {
+      MessageID msgId = generateMessageID();
+      if (!msgId.equals(other)) return msgId;
+    }
+  }
+
+  /** @return a pseudo randomly generated signature */
+  public static Signature generateSignature() {
+    byte[] signature = new byte[SIGNATURE_LENGTH];
+    RANDOM.nextBytes(signature);
+    return new Signature(signature);
+  }
+
+  /** @return a pseudo randomly generated valid Ed25519 keypair */
+  public static KeyPair generateKeyPair() {
+    byte[] privateKey = new byte[KEY_LENGTH];
+    RANDOM.nextBytes(privateKey);
+
+    Ed25519PrivateKeyParameters privateKeyParameters =
+        new Ed25519PrivateKeyParameters(privateKey, 0);
+    Ed25519PublicKeyParameters publicKeyParameters = privateKeyParameters.generatePublicKey();
+
+    return new KeyPair(
+        new PlainPrivateKey(privateKey), new PublicKey(publicKeyParameters.getEncoded()));
+  }
+
+  /** @return a pseudo randomly generated valid Ed25519 keypair */
+  public static PoPToken generatePoPToken() {
+    byte[] privateKey = new byte[KEY_LENGTH];
+    RANDOM.nextBytes(privateKey);
+
+    Ed25519PrivateKeyParameters privateKeyParameters =
+        new Ed25519PrivateKeyParameters(privateKey, 0);
+    Ed25519PublicKeyParameters publicKeyParameters = privateKeyParameters.generatePublicKey();
+
+    return new PoPToken(privateKey, publicKeyParameters.getEncoded());
```

Comment:
why not just call `generateKeyPair` above? It should be nearly the same implementation, except for the return type.
Or vice versa, let `generateKeyPair` call this.

Reply:
Yes, this has been addressed